test/test_linalg.py

```diff
@@ -3043,6 +3043,61 @@ def test_old_matrix_rank(self, device, dtype):
             self.assertEqual(torch.matrix_rank(aaT, True), np.linalg.matrix_rank(aaT.cpu().numpy(), True))
             self.assertEqual(torch.matrix_rank(aaT, 0.01, True), np.linalg.matrix_rank(aaT.cpu().numpy(), 0.01, True))
 
+    @onlyOnCPUAndCUDA
+    @dtypes(torch.float, torch.double, torch.cfloat, torch.cdouble)
+    @precisionOverride({torch.float: 1e-03, torch.cfloat: 1e-03})
+    def test_multi_dot(self, device, dtype):
```

Comment:
Add a test for shape-checking errors (e.g. first and last tensor not being 1 or 2D, intermediate tensors not being 2D)